common/src/main/java/org/dash/wallet/common/data/WalletUIConfig.kt

```diff
@@ -0,0 +1,122 @@
+    exchangeRates: ExchangeRatesProvider
+): BaseConfig(
+    context,
+    PREFERENCES_NAME,
+    walletDataProvider,
+    migrations = listOf(
+        // Migrating relevant keys from default prefs
+        SharedPreferencesMigration(
+            context = context,
+            sharedPreferencesName = context.packageName + "_preferences",
+            keysToMigrate = setOf(
+                AUTO_HIDE_BALANCE.name,
+                EXCHANGE_CURRENCY_DETECTED.name
+            )
+        ),
+        ExchangeCurrencyMigration(
+            context = context,
+            sharedPreferencesName = context.packageName + "_preferences",
+            exchangeRates = exchangeRates
+        )
+    )
+) {
+    companion object {
+        private const val PREFERENCES_NAME = "wallet_ui"
+        val AUTO_HIDE_BALANCE = booleanPreferencesKey("hide_balance")
+        val SHOW_TAP_TO_HIDE_HINT = booleanPreferencesKey("show_tap_to_hide_balance_hint")
+        val SELECTED_CURRENCY = stringPreferencesKey("exchange_currency")
+        val EXCHANGE_CURRENCY_DETECTED = booleanPreferencesKey("exchange_currency_detected")
+    }
+
+    suspend fun getExchangeCurrencyCode(): String {
+        return get(SELECTED_CURRENCY) ?: Constants.DEFAULT_EXCHANGE_CURRENCY
+    }
+
+    fun getExchangeCurrencyCodeBlocking(): String {
+        return runBlocking { getExchangeCurrencyCode() }
+    }
+}
+
+class ExchangeCurrencyMigration(
+    private val context: Context,
+    private val sharedPreferencesName: String,
+    private val exchangeRates: ExchangeRatesProvider
+) : DataMigration<Preferences> {
+
+    companion object {
+        private val log = org.slf4j.LoggerFactory.getLogger(WalletUIConfig::class.java)
+    }
+
+    private val migration = SharedPreferencesMigration(
+        context = context,
+        sharedPreferencesName = sharedPreferencesName,
+        keysToMigrate = setOf(
+            WalletUIConfig.SELECTED_CURRENCY.name
+        )
+    )
+
+    override suspend fun shouldMigrate(currentData: Preferences): Boolean {
+        return migration.shouldMigrate(currentData)
+    }
+
+    override suspend fun migrate(currentData: Preferences): Preferences {
+        try {
+            // previous versions of the app (prior to 7.3.3) may have stored an obsolete
+            // currency code in the preferences.  Let's change to the most up to date.
+            val sharedPreferences = context.getSharedPreferences(sharedPreferencesName, Context.MODE_PRIVATE)
+            val currentValue = sharedPreferences.getString(WalletUIConfig.SELECTED_CURRENCY.name, null)
+            currentValue?.let {
+                val fixedValue = CurrencyInfo.getUpdatedCurrency(CurrencyInfo.getOtherName(currentValue))
+                sharedPreferences.edit().putString(WalletUIConfig.SELECTED_CURRENCY.name, fixedValue).apply()
+            }
+            // The database might have obsolete currencies as well
+            exchangeRates.cleanupObsoleteCurrencies()
```

Comment:
This will run once on migration and ensure we never have obsolete currency names in the datastore. 
This way, we don't have to call `getOtherName` or `getUpdatedCurrency` every time.